### ISSUE TITLE
Can we enhance this project with CloudFront Functions

### ISSUE THREAD

**Issue by IamManchanda** [serverless-nextjs/serverless-next.js#1066]
**Is your feature request related to a problem? Please describe.**
No, It is instead an enhancement request

**Describe the solution you'd like**
Please have a look at 
https://aws.amazon.com/blogs/aws/introducing-cloudfront-functions-run-your-code-at-the-edge-with-low-latency-at-any-scale/

**Comment by davegariepy** [comment]
@IamManchanda possibly a use case for rewrites / redirects? I'm not sure how that would map to the _Nextjs way_ though.

**Comment by jvarho** [comment]
10KB limit means there isn't much that can be done. The route manifests alone can be larger than that and almost everything useful will require knowing which urls are pages vs. public files. Including most redirect/rewrite logic.

The only things I can think of are domain redirects and basic auth. But those are not something all sites would use and even if you do use them I'm not sure it's better to have another step when processing successful requests.

Specific sites could have many uses for them, but generic logic could not make the assumptions required to fit in the limits.

https://aws.amazon.com/cloudfront/faqs/#Limits

**Comment by dphang** [comment]
Yes, I think until it is less limited, it may not be too useful for this project (yet). And we need to first fully refactor the core code. But as mentioned it can be useful for redirects/rewrites, basic auth. And maybe for static pages, it may be useful if we can somehow compress/optimized the manifests so it can fit within the 10 KB limitation.

I will keep open for now in case of any ideas

**Comment by IamManchanda** [comment]
Cool decision @dphang ... cheers 🍻

**Comment by martg0** [comment]
I am solving the **cache-control** header **max-age=60** issue on the image optimizer component from _**_next/image**_ using a cloudfront function attached to the Viewer Response to override the cache-control header.

This is the function I am using:
`function handler(event) {
    var response = event.response;
    var headers = response.headers;
    
    // Set the cache-control header
     headers['cache-control'] = {value: 'max-age=31536000, s-maxage=31536000, public'};

    // Return response to viewers
    return response;
}`

**Comment by Vidlec** [comment]
@martg0 Are you using AWS CDK for that or is it just manually added in AWS Console?

**Comment by martg0** [comment]
@Vidlec I've added this directly on CloudFront. The issue resides when I redeploy I need to reassign the LambdaEdge Function to the Viewer Response method. Still don't know how to do this via AWS CDK. I need to investigate and learn more.
